tests/devices/test_default_clifford.py

```diff
@@ -0,0 +1,318 @@
+        phase = qml.math.divide(
+            st1, st2, out=qml.math.zeros_like(st1, dtype=complex), where=st1 != 0
+        )[qml.math.nonzero(np.round(st1, 10))]
+        if not qml.math.allclose(phase, 0.0):
+            assert qml.math.allclose(phase / phase[0], qml.math.ones(len(phase)))
+    else:
+        tableau = np.array([[0, 1, 1, 0, 0], [1, 0, 1, 1, 1], [0, 0, 0, 1, 0], [1, 0, 0, 1, 1]])
+        assert qml.math.allclose(tableau, qn_c())
+
+
+@pytest.mark.parametrize("circuit", [circuit_1])
+@pytest.mark.parametrize(
+    "meas_type",
+    ["dm", "purity"],
+)
+def test_meas_clifford(circuit, meas_type):
+    """Test that the execution of default.clifford is possible and agrees with default.qubit"""
+    dev_c = qml.device("default.clifford")
+    dev_d = qml.device("default.qubit")
+
+    def circuit_fn():
+        circuit()
+        return qml.density_matrix([0, 1]) if meas_type == "dm" else qml.purity([0, 1])
+
+    qn_c = qml.QNode(circuit_fn, dev_c)
+    qn_d = qml.QNode(circuit_fn, dev_d)
+
+    assert np.allclose(qn_c(), qn_d())
+
+
+@pytest.mark.parametrize("circuit", [circuit_1])
+def test_prep_snap_clifford(circuit):
+    """Test that the execution of default.clifford is possible and agrees with default.qubit"""
+    dev_c = qml.device("default.clifford", check_clifford=False)
+    dev_d = qml.device("default.qubit")
+
+    def circuit_fn():
+        qml.BasisState(np.array([1, 1]), wires=range(2))
+        circuit()
+        qml.Snapshot(tag="cliffy")
+        return qml.expval(qml.PauliZ(0))
+
+    qn_c = qml.QNode(circuit_fn, dev_c)
+    qn_d = qml.QNode(circuit_fn, dev_d)
+
+    assert np.allclose(qn_c(), qn_d())
+    assert np.allclose(qml.grad(qn_c)(), qml.grad(qn_d)())
+
+
+def test_max_worker_clifford():
+    """Test that the execution of default.clifford is possible and agrees with default.qubit"""
+
+    os.environ["OMP_NUM_THREADS"] = "4"
+
+    dev_c = qml.device("default.clifford", max_workers=2)
+    dev_d = qml.device("default.qubit", max_workers=2)
+
+    qscript = qml.tape.QuantumScript(
+        [qml.Hadamard(wires=[0]), qml.CNOT([0, 1])],
+        [qml.expval(qml.PauliZ(0))],
+    )
+    tapes = tuple([qscript])
+
+    program, config = dev_d.preprocess()
+    tapes, _ = program(tapes)
+    res_d = dev_d.execute(tapes, config)
+    program, config = dev_c.preprocess()
+    tapes, _ = program(tapes)
+    res_c = dev_c.execute(tapes, config)
+    assert np.allclose(res_d, res_c)
+
+    res_q = dev_c.execute(qscript, config)
+    assert np.allclose(res_q, res_c[0])
+
+    grad_t = dev_c.compute_derivatives(tapes, config)
+    grad_q = dev_c.compute_derivatives(qscript, config)
+    assert qml.math.allclose(grad_q, grad_t[0])
+
+    (res_t, grad_t) = dev_c.execute_and_compute_derivatives(tapes, config)
+    (res_q, grad_q) = dev_c.execute_and_compute_derivatives(qscript, config)
+    assert qml.math.allclose(res_q, res_t[0]) and qml.math.allclose(grad_q, grad_t[0])
+
+
+def test_tracker():
```

Comment:
Not sure if this really tests tracker. I was expecting  to test `tracker.history` etc.

Reply:
Updated the `tracker` test.